File: src/types/IK2Inventory.ts
import { IK2, IK2Body } from "./IK2";

export interface IK2InventoryBody extends IK2Body {
  folders: {
    ignore: string[];
    applies: string[];
    templates: string[];
  };
  vars?: Record<string, string>;
}

export interface IK2Inventory extends IK2<IK2InventoryBody> {}

File: types/IK2Inventory.go
package types

type IK2Inventory struct {
	K2 IK2InventoryRoot `yaml:"k2"`
}

type IK2InventoryRoot struct {
	Metadata IK2Metadata      `yaml:"metadata"`
	Body     IK2InventoryBody `yaml:"body"`
}

type IK2InventoryBody struct {
	Folders struct {
		Ignore    []string `yaml:"ignore"`
		Applies   []string `yaml:"applies"`
		Templates []string `yaml:"templates"`
	} `yaml:"folders"`
	Vars map[string]string `yaml:"vars"`
}
